File: FindACoach/src/FindACoach.Core/Domain/Entities/Activity/Activity.cs
using System.ComponentModel.DataAnnotations;

namespace FindACoach.Core.Domain.Entities.Activity
{
    public abstract class Activity
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        [StringLength(50, MinimumLength = 3)]
        public string Title { get; set; } = string.Empty;

        [StringLength(400)]
        public string? ImagePath { get; set; } = string.Empty;

        [StringLength(200)]
        public string? Description { get; set; } = string.Empty;

        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<Subject> Subjects { get; set; } = new List<Subject>();

        public Guid UserId { get; set; }
        public IdentityEntities.User? User { get; set; }

        public ICollection<Like> Likes { get; set; } = new List<Like>();

        public ICollection<Save> Saves { get; set; } = new List<Save>();

        public ICollection<Comment> Comments { get; set; } = new List<Comment>();

    }
}

File: find-a-coach-client/src/types/forum/Activity.ts
import { Comment } from "@/types/forum/Comment"

interface Activity {
  id: string;
  userId: string;
  userImagePath: string | null;
  userFirstName: string;
  userLastName: string;
  title: string;
  imagePath: string | null;
  description: string;
  createdAt: string;
  subjects: string[];
  isLiked: boolean;
  numberOfLikes: number;
  isSaved: boolean;
  comments: Comment[]
}

export type { Activity }